source/extensions/filters/network/redis_proxy/command_splitter_impl.h

```diff
@@ -234,6 +234,25 @@ class InstanceImpl : public Instance, Logger::Loggable<Logger::Id::redis> {
     std::reference_wrapper<CommandHandler> handler_;
   };
 
+  typedef std::shared_ptr<HandlerData> HandlerDataPtr;
+
+  struct HandlerEntry {
```

Comment:
Can you potentially make a single template class and share it with header map impl?

Reply:
I can make it as a follow-up if it is ok with you, I will leave a TODO for now.

Reply:
Why bother just copying the code? The template class should be easy. Please do it in this PR, or do that refactor first.

Reply:
Agree copying code and template class should be easy to add. I have not looked at `HeaderImpl` code in detail to see how easy/difficult that refactor is- that is why I was hesitant to do it in this PR. Will check and change